src/_pytest/report_log.py

```diff
@@ -0,0 +1,70 @@
+import json
+from pathlib import Path
+
+
+def pytest_addoption(parser):
+    group = parser.getgroup("terminal reporting", "report-log plugin options")
+    group.addoption(
+        "--report-log",
+        action="store",
+        metavar="path",
+        default=None,
+        help="Path to line-based json objects of test session events.",
+    )
+
+
+def pytest_configure(config):
+    report_log = config.option.report_log
+    if report_log and not hasattr(config, "slaveinput"):
+        config._report_log_plugin = ReportLogPlugin(config, Path(report_log))
+        config.pluginmanager.register(config._report_log_plugin)
+
+
+def pytest_unconfigure(config):
+    report_log_plugin = getattr(config, "_report_log_plugin", None)
+    if report_log_plugin:
+        report_log_plugin.close()
+        del config._report_log_plugin
+
+
+class ReportLogPlugin:
+    def __init__(self, config, log_path: Path):
+        self._config = config
+        self._log_path = log_path
+
+        log_path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
That should fail instead of creating folders

Reply:
Why? I think it is very convenient, plus both `junitxml` and `resultlog` create the intermediate folders as needed.